.github/actions/repo-gardening/src/tasks/add-labels/index.js

```diff
@@ -0,0 +1,250 @@
+	debug( 'add-labels: Get list of files modified in this PR.' );
+
+	for await ( const response of octokit.paginate.iterator( octokit.pulls.listFiles, {
+		owner,
+		repo,
+		pull_number: +number,
+	} ) ) {
+		response.data.map( file => {
+			fileList.push( file.filename );
+		} );
+	}
+
+	return fileList;
+}
+
+/**
+ * Clean up a feature name:
+ * - Handle some exceptions in our codename / feature names.
+ * - Replace dashes by spaces.
+ * - Capitalize.
+ *
+ * @param {string} name - Feature name.
+ *
+ * @returns {string} Cleaned up feature name.
+ */
+function cleanName( name ) {
+	// Sharedaddy is a legacy codename.
+	if ( name === 'sharedaddy' ) {
+		name = 'Sharing';
+	}
+
+	// Our Shortcodes feature includes shortcodes and embeds.
+	if ( name === 'shortcodes' ) {
+		name = 'Shortcodes / Embeds';
+	}
+
+	// We name our CPTs "Custom Content Types" to avoid confusion with WordPress's CPT.
+	if ( name === 'custom-post-types' ) {
+		name = 'Custom Content Types';
+	}
+
+	// Our widgets are "Extra Sidebar Widgets".
+	if ( name === 'widgets' ) {
+		name = 'Extra Sidebar Widgets';
+	}
+
+	// Simple Payments was renamed into "Pay With Paypal".
+	if ( name === 'simple-payments' ) {
+		name = 'Pay With Paypal';
+	}
+
+	// WordPress.com Block Editor lives under 'wpcom-block-editor'.
+	if ( name === 'wpcom-block-editor' ) {
+		name = 'WordPress.com Block Editor';
+	}
+
+	// WordAds is a codename. We name the feature just "Ad" or "Ads".
+	if ( name === 'wordads' ) {
+		name = 'Ad';
+	}
+
+	// Latest Instagram Posts used to be named Instagram Gallery.
+	if ( name === 'instagram-gallery' ) {
+		name = 'Latest Instagram Posts';
+	}
+
+	// Payments used to be called Recurring Payments.
+	if ( name === 'recurring-payments' ) {
+		name = 'Payments';
+	}
+
+	// Rating Star was renamed into Star Rating.
+	if ( name === 'rating-star' ) {
+		name = 'Star Rating';
+	}
+
+	return (
+		name
+			// Break up words
+			.split( '-' )
+			// Capitalize first letter of each word.
+			.map( word => `${ word[ 0 ].toUpperCase() }${ word.slice( 1 ) }` )
+			// Spaces between words.
+			.join( ' ' )
+	);
+}
+
+/**
+ * Build a list of labels to add to the issue, based off our file list.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - PR number.
+ *
+ * @returns {Promise<Array>} Promise resolving to an array of keywords we'll search for.
+ */
+async function getLabelsToAdd( octokit, owner, repo, number ) {
+	const keywords = new Set();
+
+	// Get next valid milestone.
+	const files = await getFiles( octokit, owner, repo, number );
+
+	if ( ! files ) {
+		throw new Error( 'No files were modified in this PR' );
+	}
+
+	debug( 'add-labels: Loop through all files modified in this PR and add matching labels.' );
+
+	files.map( file => {
+		// Projects.
+		const project = file.match( /^projects\/(?<ptype>[a-zA-Z-]*)\/(?<pname>[a-zA-Z-]*)\// );
```

Comment:
I see you changed `\w` to `[a-zA-Z-]` to account for lazy-images. But what about if we get something with numbers at some point, or something else? Maybe just go all the way to `[^/]` to accept anything anyone might use in a path component.